packages/next-on-pages/src/buildApplication/buildWorkerFile.ts

```diff
@@ -79,5 +79,38 @@ export async function buildWorkerFile(
 		minify,
 	});
 
+	await build({
+		entryPoints: ['adaptor.ts', 'cache-api.ts', 'kv.ts'].map(fileName =>
+			join(templatesDir, 'cache', fileName),
+		),
+		bundle: false,
+		target: 'es2022',
+		platform: 'neutral',
+		outdir: join(nopDistDir, 'cache'),
+		minify,
+		plugins: [
+			{
+				name: 'fix-import-extension-for-unbundled-adaptors',
+				setup(build) {
+					build.onLoad({ filter: /.*/ }, async args => {
+						const contents = await readFile(args.path, 'utf8');
+						const newContents = contents.replace(
+							"from './adaptor';",
```

Comment:
could you add `.js` in the ts template code instead?

Reply:
looks like it 👍🏻